dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/NoneFunctionChoiceBehavior.cs

```diff
@@ -0,0 +1,106 @@
+    /// List of the functions that the model can choose from.
+    /// </summary>
+    private readonly IEnumerable<KernelFunction>? _functions;
+
+    /// <summary>
+    /// This class type discriminator used for polymorphic deserialization of the type specified in JSON and YAML prompts.
+    /// </summary>
+    public const string TypeDiscriminator = "none";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NoneFunctionChoiceBehavior"/> class.
+    /// </summary>
+    [JsonConstructor]
+    public NoneFunctionChoiceBehavior()
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NoneFunctionChoiceBehavior"/> class.
+    /// </summary>
+    /// <param name="functions">The subset of the <see cref="Kernel"/>'s plugins' functions to provide to the model.
+    /// If null or empty, all <see cref="Kernel"/>'s plugins' functions are provided to the model.</param>
+    public NoneFunctionChoiceBehavior(IEnumerable<KernelFunction> functions)
+    {
+        this._functions = functions;
+        this.Functions = functions.Select(f => FunctionName.ToFullyQualifiedName(f.Name, f.PluginName)).ToList();
+    }
+
+    /// <summary>
+    /// Fully qualified names of subset of the <see cref="Kernel"/>'s plugins' functions to provide to the model.
+    /// If null or empty, all <see cref="Kernel"/>'s plugins' functions are provided to the model.
+    /// </summary>
+    [JsonPropertyName("functions")]
+    [JsonConverter(typeof(FunctionNameFormatJsonConverter))]
+    public IList<string>? Functions { get; set; }
+
+    /// <inheritdoc/>
+    public override FunctionChoiceBehaviorConfiguration GetConfiguration(FunctionChoiceBehaviorContext context)
+    {
+        List<KernelFunction>? availableFunctions = null;
+
+        // Handle functions provided via the 'Functions' property as function fully qualified names.
+        if (this.Functions is { } functionFQNs && functionFQNs.Any())
+        {
+            availableFunctions = [];
+
+            foreach (var functionFQN in functionFQNs)
+            {
+                var nameParts = FunctionName.Parse(functionFQN);
+
+                // Check if the function is available in the kernel.
+                if (context.Kernel!.Plugins.TryGetFunction(nameParts.PluginName, nameParts.Name, out var function))
+                {
+                    availableFunctions.Add(function);
+                    continue;
+                }
+
+                // Check if a function instance that was not imported into the kernel was provided through the constructor.
+                function = this._functions?.FirstOrDefault(f => f.Name == nameParts.Name && f.PluginName == nameParts.PluginName);
+                if (function is not null)
+                {
+                    availableFunctions.Add(function);
+                    continue;
+                }
+
+                throw new KernelException($"The specified function {functionFQN} is not available.");
```

Comment:
Syntax thing: We will never know if the function is available or not, but we surely knows we didn't find it. 

```suggestion
                throw new KernelException($"The specified function {functionFQN} was not found.");
```

Reply:
fixed